dart-impl/base/include/dash/dart/base/atomic.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * \file atomic.h
+ *
+ * Provides atomic operations on basic integer data types.
+ * Feel free to add operations you may require.
+ *
+ * See the URL below for details on the operations:
+ * https://gcc.gnu.org/onlinedocs/gcc/_005f_005fsync-Builtins.html#g_t_005f_005fsync-Builtins
+ *
+ * See also the file kmp_os.h of the Clang OpenMP library, which served as inspiration.
+ *
+ * TODO: Check that all compilers support these intrinsics, esp. Cray?
+ */
+
+#ifndef DASH_DART_BASE_ATOMIC_H_
+#define DASH_DART_BASE_ATOMIC_H_
+
+
+#define DART_FETCH_AND_ADD64(ptr, val)   __sync_fetch_and_add((int64_t *)(ptr), (val))
```

Comment:
I agree! We def' do not want to ship assembly (If we do some day: I'm actually way better at ARM asm than x86)